libs/designsystem/src/lib/components/calendar/calendar.component.ts

```diff
@@ -399,32 +421,58 @@ export class CalendarComponent implements OnInit, AfterViewInit, OnChanges {
   }
 
   changeMonth(index: number) {
-    this.changeActiveView(index, 'months');
+    this.changeActiveView(index, TimeUnit.month);
   }
 
   public changeYear(year: number) {
-    this.changeActiveView(year - this.activeMonth.year(), 'years');
+    this.changeActiveView(year - getYear(this.activeMonth), TimeUnit.year);
   }
 
-  private changeActiveView(index: number, unit: moment.unitOfTime.Base) {
-    if (index != 0) {
-      this.activeMonth.add(index, unit);
-      this.refreshActiveMonth();
+  private changeActiveView(index: number, unit: TimeUnit) {
+    if (index === 0) return;
+    switch (unit) {
+      case TimeUnit.year:
+        this.activeMonth = addYears(this.activeMonth, index);
+        break;
+
+      case TimeUnit.month:
+        this.activeMonth = addMonths(this.activeMonth, index);
+        break;
+
+      case TimeUnit.day:
+        this.activeMonth = addDays(this.activeMonth, index);
+        break;
+
+      case TimeUnit.hour:
+        this.activeMonth = addHours(this.activeMonth, index);
+        break;
```

Comment:
Why not use `date-fns`'s `add`method? 🤔 
https://date-fns.org/v2.22.1/docs/add

Reply:
yes, why indeed not 😄  changing